test/unit/test-cid.js

```diff
@@ -856,6 +878,88 @@ describes.realWin('cid', {amp: true}, (env) => {
     }
   );
 
+  describe('CID backup', () => {
+    beforeEach(() => {
+      cid.isBackupCidExpOn = true;
+    });
+
+    it('generate new CID and backup', async () => {
+      env.sandbox
+        .stub(cid.viewerCidApi_, 'isSupported')
+        .returns(Promise.resolve(false));
+      storageGetUnexpiredStub
+        .withArgs('amp-cid:foo', duration)
+        .returns(Promise.resolve());
+      setCookie(win, 'foo', '', 0);
+      const fooCid = await cid.get(
+        {
+          scope: 'foo',
+          createCookieIfNotPresent: true,
+        },
+        hasConsent
+      );
+      expect(fooCid).to.have.string('amp-');
+      expect(storageSetStub).to.be.calledWith('amp-cid:foo', fooCid);
+    });
+
+    it('should find AMP generated CID in cookie and backup', async () => {
+      env.sandbox
+        .stub(cid.viewerCidApi_, 'isSupported')
+        .returns(Promise.resolve(false));
+      const cidString = 'amp-abc123';
+      win.document.cookie = `foo=${cidString};`;
+
+      const fooCid = await cid.get(
+        {
+          scope: 'foo',
+          createCookieIfNotPresent: true,
+        },
+        hasConsent
+      );
+      expect(fooCid).to.equal(cidString);
+      expect(storageSetStub).to.be.calledWith('amp-cid:foo', fooCid);
+
+      storageSetStub.resetHistory();
+      const nonAmpCidString = 'xyz987';
+      win.document.cookie = `bar=${nonAmpCidString};`;
+      expect(
+        await cid.get(
+          {
+            scope: 'bar',
+            createCookieIfNotPresent: true,
+          },
+          hasConsent
+        )
+      ).to.equal(nonAmpCidString);
+      expect(storageSetStub).to.not.be.called;
+    });
+
+    it('should use CID backup', async () => {
```

Comment:
could you please add three more tests

1. cookie value and localStorage value are different. cookie value should be used and overwrite localStorage value.

2. localStorage value is out of data. A new value should be created and write to cookie and localStorage.

3. localStorage value should be not used, and not updated when user opt-out